cpp/core/jni/JniWrapper.cc

```diff
@@ -159,7 +154,7 @@ class JavaInputStreamAdaptor final : public arrow::io::InputStream {
 
   arrow::Result<std::shared_ptr<arrow::Buffer>> Read(int64_t nbytes) override {
     GLUTEN_ASSIGN_OR_THROW(auto buffer, arrow::AllocateResizableBuffer(nbytes, defaultArrowMemoryPool().get()))
```

Comment:
Should we use ArrowMemoryPool with listenable allocator here?

Reply:
Agreed. The explicit calls to default Velox/Arrow pools should be finally reviewed and optimized from our code base.